controllers/tools.js

```diff
@@ -111,18 +111,24 @@ const getDetailHandler = (req, res, next) => {
 const showToolHandler = (req, res, next) => {
 	const context = req.originalUrl.split('/')[1];
 
-	Promise.all([
-		api(req).get(`/ltiTools/${req.params.ltiToolId}`),
-		api(req).get(`/${context}/${req.params.courseId}`),
-	])
-		.then(([tool, course]) => {
-			const renderPath = tool.isLocal ? 'courses/run-tool-local' : 'courses/run-lti';
-			res.render(renderPath, {
-				course,
-				title: `${tool.name}, Kurs/Fach: ${course.name}`,
-				tool,
-			});
+	Promise.all((req.params.courseId
+		? [
+			api(req).get(`/ltiTools/${req.params.ltiToolId}`),
+			api(req).get(`/${context}/${req.params.courseId}`),
+		]
+		: [
+			api(req).get('/ltiTools/', { qs: { friendlyUrl: req.params.ltiToolId } }),
+			Promise.resolve({ name: '' }),
+		]
+	)).then(([tool, course]) => {
+		tool = (req.params.courseId ? tool : tool.data[0]);
+		const renderPath = tool.isLocal ? 'courses/run-tool-local' : 'courses/run-lti';
```

Comment:
Without additional seed data/migration I get `UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'isLocal' of undefined` here. Please catch undefined errors here and provide seed data to make that feature available locally.

Reply:
Added error handling